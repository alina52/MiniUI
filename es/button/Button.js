"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _icon = require("../icon");

function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = (0, _getPrototypeOf2.default)(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = (0, _getPrototypeOf2.default)(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2.default)(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Date.prototype.toString.call(Reflect.construct(Date, [], function () {})); return true; } catch (e) { return false; } }

var types = {
  success: "success",
  error: "error",
  warning: "warning",
  default: "default",
  primary: "primary",
  link: "link",
  info: "info"
};
var sizes = {
  small: "small",
  default: "default",
  large: "large"
};

var Button = /*#__PURE__*/function (_Component) {
  (0, _inherits2.default)(Button, _Component);

  var _super = _createSuper(Button);

  function Button() {
    (0, _classCallCheck2.default)(this, Button);
    return _super.apply(this, arguments);
  }

  (0, _createClass2.default)(Button, [{
    key: "render",
    value: function render() {
      var _cls3;

      var _this$props = this.props,
          type = _this$props.type,
          children = _this$props.children,
          prefixCls = _this$props.prefixCls,
          disabled = _this$props.disabled,
          block = _this$props.block,
          onClick = _this$props.onClick,
          href = _this$props.href,
          dashed = _this$props.dashed,
          loading = _this$props.loading,
          hollow = _this$props.hollow,
          size = _this$props.size,
          circle = _this$props.circle,
          className = _this$props.className,
          attr = (0, _objectWithoutProperties2.default)(_this$props, ["type", "children", "prefixCls", "disabled", "block", "onClick", "href", "dashed", "loading", "hollow", "size", "circle", "className"]);
      var isDisabled = disabled ? {
        disabled: true
      } : {
        onClick: onClick
      };

      if (href) {
        return /*#__PURE__*/_react.default.createElement("a", (0, _extends2.default)({
          href: disabled ? "javascript:void(0);" : href,
          disabled: disabled,
          className: (0, _classnames.default)("".concat(prefixCls, "-link"), (0, _defineProperty2.default)({}, "".concat(prefixCls, "-link-disabled"), disabled))
        }, attr), children);
      }

      return /*#__PURE__*/_react.default.createElement("div", {
        className: (0, _classnames.default)(prefixCls, (0, _defineProperty2.default)({}, "".concat(prefixCls, "-block"), block))
      }, /*#__PURE__*/_react.default.createElement("button", (0, _extends2.default)({
        type: "button"
      }, isDisabled, {
        className: (0, _classnames.default)("".concat(prefixCls, "-btn"), className, "".concat(prefixCls, "-btn-").concat(type), (_cls3 = {
          'btn-hollow': hollow,
          'btn-disabled': disabled,
          'btn-block': block,
          'btn-dashed': dashed,
          'btn-loading': loading
        }, (0, _defineProperty2.default)(_cls3, "btn-size-".concat(size), size !== sizes.default), (0, _defineProperty2.default)(_cls3, 'btn-circle', circle), _cls3))
      }, attr), !circle && loading && /*#__PURE__*/_react.default.createElement(_icon.LoadingIcon, {
        className: "".concat(prefixCls, "-loading")
      }), /*#__PURE__*/_react.default.createElement("span", null, children)));
    }
  }]);
  return Button;
}(_react.Component);

(0, _defineProperty2.default)(Button, "propTypes", {
  prefixCls: _propTypes.default.string.isRequired,
  type: _propTypes.default.oneOf(Object.values(types)),
  disabled: _propTypes.default.bool,
  block: _propTypes.default.bool,
  onClick: _propTypes.default.func,
  href: _propTypes.default.string,
  dashed: _propTypes.default.bool,
  loading: _propTypes.default.bool,
  hollow: _propTypes.default.bool,
  circle: _propTypes.default.bool,
  size: _propTypes.default.oneOf(Object.values(sizes))
});
(0, _defineProperty2.default)(Button, "defaultProps", {
  prefixCls: "mini-ui-button",
  type: types.default,
  disabled: false,
  href: "",
  dashed: false,
  block: false,
  loading: false,
  hollow: false,
  circle: false,
  size: sizes.default,
  onClick: function onClick() {}
});
var _default = Button;
exports.default = _default;