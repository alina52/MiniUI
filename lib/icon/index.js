"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.PasswordHideIcon = exports.PasswordShowIcon = exports.UserIcon = exports.EmptyIcon = exports.CloseCircleIcon = exports.CalendarIcon = exports.FileUploadIcon = exports.ArrowLeftIcon = exports.ArrowRightIcon = exports.DownIcon = exports.UpIcon = exports.CloseIcon = exports.WarningIcon = exports.ErrorIcon = exports.SuccessIcon = exports.LoadingIcon = exports.InfoIcon = void 0;

var _react = _interopRequireDefault(require("react"));

var _fi = require("react-icons/fi");

var _io = require("react-icons/io");

var InfoIcon = function InfoIcon(props) {
  return /*#__PURE__*/_react.default.createElement(_fi.FiVolume2, props);
};

exports.InfoIcon = InfoIcon;

var LoadingIcon = function LoadingIcon(props) {
  return /*#__PURE__*/_react.default.createElement(_fi.FiLoader, props);
};

exports.LoadingIcon = LoadingIcon;

var SuccessIcon = function SuccessIcon(props) {
  return /*#__PURE__*/_react.default.createElement(_fi.FiCheckCircle, props);
};

exports.SuccessIcon = SuccessIcon;

var ErrorIcon = function ErrorIcon(props) {
  return /*#__PURE__*/_react.default.createElement(_fi.FiXCircle, props);
};

exports.ErrorIcon = ErrorIcon;

var WarningIcon = function WarningIcon(props) {
  return /*#__PURE__*/_react.default.createElement(_fi.FiAlertCircle, props);
};

exports.WarningIcon = WarningIcon;

var CloseIcon = function CloseIcon(props) {
  return /*#__PURE__*/_react.default.createElement(_fi.FiX, props);
};

exports.CloseIcon = CloseIcon;

var UpIcon = function UpIcon(props) {
  return /*#__PURE__*/_react.default.createElement(_fi.FiChevronUp, props);
};

exports.UpIcon = UpIcon;

var DownIcon = function DownIcon(props) {
  return /*#__PURE__*/_react.default.createElement(_fi.FiChevronDown, props);
};

exports.DownIcon = DownIcon;

var ArrowRightIcon = function ArrowRightIcon(props) {
  return /*#__PURE__*/_react.default.createElement(_fi.FiChevronRight, props);
};

exports.ArrowRightIcon = ArrowRightIcon;

var ArrowLeftIcon = function ArrowLeftIcon(props) {
  return /*#__PURE__*/_react.default.createElement(_fi.FiChevronLeft, props);
};

exports.ArrowLeftIcon = ArrowLeftIcon;

var FileUploadIcon = function FileUploadIcon(props) {
  return /*#__PURE__*/_react.default.createElement(_fi.FiUploadCloud, props);
};

exports.FileUploadIcon = FileUploadIcon;

var CalendarIcon = function CalendarIcon(props) {
  return /*#__PURE__*/_react.default.createElement(_fi.FiCalendar, props);
};

exports.CalendarIcon = CalendarIcon;

var CloseCircleIcon = function CloseCircleIcon(props) {
  return /*#__PURE__*/_react.default.createElement(_io.IoMdCloseCircle, props);
};

exports.CloseCircleIcon = CloseCircleIcon;

var EmptyIcon = function EmptyIcon(props) {
  return /*#__PURE__*/_react.default.createElement(_fi.FiHardDrive, props);
};

exports.EmptyIcon = EmptyIcon;

var UserIcon = function UserIcon(props) {
  return /*#__PURE__*/_react.default.createElement(_fi.FiUser, props);
};

exports.UserIcon = UserIcon;

var PasswordShowIcon = function PasswordShowIcon(props) {
  return /*#__PURE__*/_react.default.createElement(_fi.FiEye, props);
};

exports.PasswordShowIcon = PasswordShowIcon;

var PasswordHideIcon = function PasswordHideIcon(props) {
  return /*#__PURE__*/_react.default.createElement(_fi.FiEyeOff, props);
};

exports.PasswordHideIcon = PasswordHideIcon;