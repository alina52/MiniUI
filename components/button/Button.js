import React, { Component } from "react";
import PropTypes from "prop-types";
import cls from "classnames";
import { LoadingIcon } from "../icon";
const types = {
  success: "success",
  error: "error",
  warning: "warning",
  default: "default",
  primary: "primary",
  link: "link",
  info: "info"
};

const sizes = {
  small: "small",
  default: "default",
  large: "large"
};

class Button extends Component {
  static propTypes = {
    prefixCls: PropTypes.string.isRequired,
    type: PropTypes.oneOf(Object.values(types)),
    disabled: PropTypes.bool,
    block: PropTypes.bool,
    onClick: PropTypes.func,
    href: PropTypes.string,
    dashed: PropTypes.bool,
    loading: PropTypes.bool,
    hollow: PropTypes.bool,
    circle: PropTypes.bool,
    size: PropTypes.oneOf(Object.values(sizes))
  };
  static defaultProps = {
    prefixCls: "mini-ui-button",
    type: types.default,
    disabled: false,
    href: "",
    dashed: false,
    block: false,
    loading: false,
    hollow: false,
    circle: false,
    size: sizes.default,
    onClick: () => {}
  };
  render() {
    const {
      type,
      children,
      prefixCls,
      disabled,
      block,
      onClick,
      href,
      dashed,
      loading,
      hollow,
      size,
      circle,
      className,
      ...attr
    } = this.props;
    const isDisabled = disabled ? { disabled: true } : { onClick };
    if (href) {
      return (
        <a
          href={disabled ? "javascript:void(0);" : href}
          disabled={disabled}
          className={cls(`${prefixCls}-link`,{[`${prefixCls}-link-disabled`]:disabled})}
          {...attr}
        >
          {children}
        </a>
      );
    }
    return (
      <div className={cls(prefixCls, {[`${prefixCls}-block`]:block})}>
        <button
          type="button"
          {...isDisabled}
          className={cls(`${prefixCls}-btn`,className,`${prefixCls}-btn-${type}`,
          {
            'btn-hollow': hollow,
            'btn-disabled':disabled,
            'btn-block':block,
            'btn-dashed':dashed,
            'btn-loading':loading,
            [`btn-size-${size}`]: size !== sizes.default,
            ['btn-circle']: circle
          }
        )}
          {...attr}
        >
          {
            !circle && loading && <LoadingIcon className={`${prefixCls}-loading`}/>
          }
          <span>{children}</span>
        </button>
      </div>
    );
  }
}

export default Button;
