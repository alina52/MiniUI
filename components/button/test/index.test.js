import React from "react";
import assert from "power-assert";
import { render,shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Button from "../index";
import { LoadingIcon } from "../../icon";

describe("<Button/>", () => {
  it("should render a <Button/> components", () => {
    const wrapper = render(
      <div>
        <Button>Click</Button>
        <Button type="primary">Click</Button>
        <Button type="info">Click</Button>
        <Button type="warning">Click</Button>
        <Button type="error">Click</Button>
        <Button type="success">Click</Button>
        <Button disabled>Click</Button>
        <Button loading>Loading..</Button>
        <Button block>100%</Button>
        <Button circle>Circle</Button>
        <Button type="primary" size="large"> Large </Button>
        <Button type="info">Middle</Button>
        <Button type="warning" size="small"> small </Button>
      </div>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should can trigger click event", () => {
    const onClick = jest.fn();
    const wrapper = shallow(<Button onClick={onClick}>Click</Button>);
    wrapper.find("button").simulate("click");
    expect(onClick).toHaveBeenCalled();
  });

  it("should can not trigger click event", () => {
    const onClick = jest.fn();
    const wrapper = shallow(
      <Button onClick={onClick} disabled>
        Disable
      </Button>
    );
    // simulate
    wrapper.find("button").simulate("click");
    expect(onClick).not.toHaveBeenCalled();
  });

  it("should can not render loading icon when mode is circle", () => {
    const wrapper = shallow(
      <Button loading circle>
        mini-ui
      </Button>
    );
    assert(wrapper.find(LoadingIcon).length === 0);
    assert(wrapper.find(".mini-ui-button-loading").length === 0);
  });

  it("should render link", () => {
    const wrapper = shallow(<Button href="/">mini-ui</Button>);
    assert(wrapper.find(".mini-ui-button-link").length >= 1);
  });

  it("should can not click link when disabled", () => {
    const onClick = jest.fn();
    const wrapper = shallow(
      <Button href="#" onClick={onClick} disabled>
        mini-ui
      </Button>
    );
    wrapper.find("a").simulate("click");
    expect(onClick).not.toHaveBeenCalled();
  });
});
