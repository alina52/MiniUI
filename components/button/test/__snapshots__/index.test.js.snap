// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> should render a <Button/> components 1`] = `
<div>
  <div
    class="mini-ui-button"
  >
    <button
      class="mini-ui-button-btn mini-ui-button-btn-default"
      type="button"
    >
      <span>
        Click
      </span>
    </button>
  </div>
  <div
    class="mini-ui-button"
  >
    <button
      class="mini-ui-button-btn mini-ui-button-btn-primary"
      type="button"
    >
      <span>
        Click
      </span>
    </button>
  </div>
  <div
    class="mini-ui-button"
  >
    <button
      class="mini-ui-button-btn mini-ui-button-btn-info"
      type="button"
    >
      <span>
        Click
      </span>
    </button>
  </div>
  <div
    class="mini-ui-button"
  >
    <button
      class="mini-ui-button-btn mini-ui-button-btn-warning"
      type="button"
    >
      <span>
        Click
      </span>
    </button>
  </div>
  <div
    class="mini-ui-button"
  >
    <button
      class="mini-ui-button-btn mini-ui-button-btn-error"
      type="button"
    >
      <span>
        Click
      </span>
    </button>
  </div>
  <div
    class="mini-ui-button"
  >
    <button
      class="mini-ui-button-btn mini-ui-button-btn-success"
      type="button"
    >
      <span>
        Click
      </span>
    </button>
  </div>
  <div
    class="mini-ui-button"
  >
    <button
      class="mini-ui-button-btn mini-ui-button-btn-default btn-disabled"
      disabled=""
      type="button"
    >
      <span>
        Click
      </span>
    </button>
  </div>
  <div
    class="mini-ui-button"
  >
    <button
      class="mini-ui-button-btn mini-ui-button-btn-default btn-loading"
      type="button"
    >
      <svg
        class="mini-ui-button-loading"
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          x1="12"
          x2="12"
          y1="2"
          y2="6"
        />
        <line
          x1="12"
          x2="12"
          y1="18"
          y2="22"
        />
        <line
          x1="4.93"
          x2="7.76"
          y1="4.93"
          y2="7.76"
        />
        <line
          x1="16.24"
          x2="19.07"
          y1="16.24"
          y2="19.07"
        />
        <line
          x1="2"
          x2="6"
          y1="12"
          y2="12"
        />
        <line
          x1="18"
          x2="22"
          y1="12"
          y2="12"
        />
        <line
          x1="4.93"
          x2="7.76"
          y1="19.07"
          y2="16.24"
        />
        <line
          x1="16.24"
          x2="19.07"
          y1="7.76"
          y2="4.93"
        />
      </svg>
      <span>
        Loading..
      </span>
    </button>
  </div>
  <div
    class="mini-ui-button mini-ui-button-block"
  >
    <button
      class="mini-ui-button-btn mini-ui-button-btn-default btn-block"
      type="button"
    >
      <span>
        100%
      </span>
    </button>
  </div>
  <div
    class="mini-ui-button"
  >
    <button
      class="mini-ui-button-btn mini-ui-button-btn-default btn-circle"
      type="button"
    >
      <span>
        Circle
      </span>
    </button>
  </div>
  <div
    class="mini-ui-button"
  >
    <button
      class="mini-ui-button-btn mini-ui-button-btn-primary btn-size-large"
      type="button"
    >
      <span>
         Large 
      </span>
    </button>
  </div>
  <div
    class="mini-ui-button"
  >
    <button
      class="mini-ui-button-btn mini-ui-button-btn-info"
      type="button"
    >
      <span>
        Middle
      </span>
    </button>
  </div>
  <div
    class="mini-ui-button"
  >
    <button
      class="mini-ui-button-btn mini-ui-button-btn-warning btn-size-small"
      type="button"
    >
      <span>
         small 
      </span>
    </button>
  </div>
</div>
`;
